packages/flutter_tools/test/commands.shard/hermetic/build_windows_test.dart

```diff
@@ -535,6 +540,145 @@ if %errorlevel% neq 0 goto :VCEnd</Command>
     FeatureFlags: () => TestFeatureFlags(isWindowsEnabled: true),
   });
 
+  testUsingContext("Windows build uses pubspec's version", () async {
+    final FakeVisualStudio fakeVisualStudio = FakeVisualStudio();
+    final BuildWindowsCommand command = BuildWindowsCommand()
+      ..visualStudioOverride = fakeVisualStudio;
+    setUpMockProjectFilesForBuild();
+
+    fileSystem.file('pubspec.yaml')
+      ..createSync()
+      ..writeAsStringSync('version: 2.3.4+5');
```

Comment:
It's a bit odd to use a version that is explicitly discouraged by Dart docs (which say not to use `+x` after 1.0.0) as the test here.

Reply:
I'd prefer to use non-zero and non-default values for testing purposes. My understanding is `+x` is perfectly valid after 1.0.0, it's just not recommended.

Here's what I found from Dart's docs: https://dart.dev/tools/pub/versioning#semantic-versions

> Although semantic versioning doesn’t promise any compatibility between versions prior to 1.0.0, the Dart community convention is to treat those versions semantically as well. The interpretation of each number is just shifted down one slot: going from 0.1.2 to 0.2.0 indicates a breaking change, going to 0.1.3 indicates a new feature, and going to 0.1.2+1 indicates a change that doesn’t affect the public API. For simplicity’s sake, avoid using + after the version reaches 1.0.0

Reply:
I don't follow; why do you want to test cases that don't match things that people are likely to ever do, rather than have two tests, one for `x.y.z` and one for `0.x.y+z` so that our tests actually test that common cases will work?

Reply:
The goal of this test is to verify that Windows build passes the version information along. I'd like to use "marker values" that are clearly controlled by the test. For example, imagine my pubspec value was `0.1.0+1` - it's not evident if the resulting cmake values are simply default/fallback values or values from my test's input. A version like `2.3.4+5` - while perhaps not recommended - is clearly the result of my test's input.

> why do you want to test cases that don't match things that people are likely to ever do

This case is common, there are [>100 hits for `1.x.y+z` on GitHub](https://cs.github.com/?scopeName=All+repos&scope=&q=%2Fversion%3A%5Cs*1.%5B.0-9A-Za-z%5D%2B%5C%2B%2F+path%3Apubspec.yaml+) alone.

Reply:
> For example, imagine my pubspec value was `0.1.0+1`

I agree that would be a very poor example; it's also not at all what I suggested.

> This case is common, there are [>100 hits for `2.x.y+z` on GitHub](https://cs.github.com/?scopeName=All+repos&scope=&q=%2Fversion%3A%5Cs*2.%5B.0-9A-Za-z%5D%2B%5C%2B%2F+path%3Apubspec.yaml+) alone.

I don't have access to that search system, but can pretty much guarantee that `<something-non-zero>.x.y` without a build number is much, much more common, and currently you **are not testing** that case. And per the other comment thread, the original version of this PR handled that incorrectly by treating it as equivalent to `+1`.

Can you explain why you don't want to add a test case that covers the most common versioning scheme and that would have exposed a bug that this PR already had?

Reply:
@stuartmorgan and I had a chat as there was some miscommunication. We decided that:

1. We don't need complete coverage from integration tests, however...
2. We will have integration tests for both `a.b.c` and `a.b.c+d` cases as they are common cases
3. I will take a closer look at installers/auto updaters to see whether we should flow the build number to the Windows version numbers as well as the version string.

Reply:
Marking this as resolved as I added the integration tests and flowed the build number to the Windows version number. Please let me know if you have additional feedback!